### ISSUE TITLE
[disallow-type-imports] Disallow the Type import syntax

### ISSUE THREAD

**Issue by remcohaszing** [typescript-eslint/typescript-eslint#2406]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Additional Info**

TypeScript 3.8 introduced type imports and type exports. While this seemed great at first, I concluded it’s not as great as it seems. Type exports help transpilers such as Babel or `ts-loader` without the type checker to remove these exports. Type imports have no real added value.

TypeScript has the option to fail throw an error if an import can be transformed into a type import, but not the other way around.

I would like to propose a rule to disallow type imports. They will be autofixed to regular imports.

**Bad**

```ts
import type { Foo } from 'bar';
```

**Good**

```ts
import { Foo } from 'bar';
```

I am willing to create a PR for this.

**Comment by bradzacher** [comment]
[`consistent-type-imports`](https://github.com/typescript-eslint/typescript-eslint/blob/v4/packages/eslint-plugin/docs/rules/consistent-type-imports.md) will be released with our v4 release.

You can try it out using the `rc-v4` tag on NPM.
